### PR TITLE
bug(source-close-com): add array type support to custom field types

### PR DESCRIPTION
**Pull request by jtruty**
## What
Fixes a sync error we are seeing with some Close.com custom fields:
```
Failed to transform value ['My custom field value'] of type 'array' to '['null', 'string', 'number', 'boolean']', key path: '.custom.lcf_xxxxxxxx'
```
## How
Adds array type as an allowed type for custom fields.

## Review guide
1. [airbyte-integrations/connectors/source-close-com/source_close_com/source.py](https://github.com/airbytehq/airbyte/compare/master...quotapath:airbyte:jtruty/source-close-com-custom-field-arrays?expand=1#diff-06ed6865598c2c4c023afaa3742dedf502b080ce99a2d4787e399640cd826d01)

## User Impact
User can use array values in custom fields.

## Can this PR be safely reverted and rolled back?

- [X ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/source-close-com/pyproject.toml [hunk 1]
```diff
@@ -3,7 +3,7 @@ requires = [ "poetry-core>=1.0.0",]
 build-backend = "poetry.core.masonry.api"
 
 [tool.poetry]
-version = "0.5.3"
+version = "0.5.4"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-close-com/pyproject.toml:6]: @jtruty can you run `poetry lock` in this path to regenerate the dependency file? Also please update the `docs/integrations/source-close` changelog documentation.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

**Comment by marcosmarxm** [comment]
@jtruty did you check my last comment?

File: airbyte-integrations/connectors/source-close-com/pyproject.toml (hunk 1 above)
Comment (jtruty) [reply airbyte-integrations/connectors/source-close-com/pyproject.toml:6]: done!

File: docs/integrations/sources/close-com.md [hunk 2]
```diff
@@ -109,6 +109,7 @@ The Close.com connector is subject to rate limits. For more information on this
 
 | Version | Date       | Pull Request                                             | Subject                                                                                                |
 | :------ | :--------- | :------------------------------------------------------- | :----------------------------------------------------------------------------------------------------- |
+| 0.5.3 | 2024-07-01 | [40515](https://github.com/airbytehq/airbyte/pull/40515) | Add support for array type custom fields |
```
Comment (marcosmarxm) [inline docs/integrations/sources/close-com.md:112]:
```suggestion
| 0.5.4 | 2024-07-01 | [40515](https://github.com/airbytehq/airbyte/pull/40515) | Add support for array type custom fields |
```

@jtruty need you to fix this small typo. I don't have permission to edit your branch.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Sorry one missing typo.

File: docs/integrations/sources/close-com.md (hunk 2 above)
Comment (jtruty) [reply docs/integrations/sources/close-com.md]: done!

**Review by marcosmarxm — APPROVED** [review]

**Comment by marcosmarxm** [comment]
Hello your feedback matters a lot to us. Can you spare just 2 minutes to complete a [survey](https://form.typeform.com/to/DOBqejte)? We’re dedicated to making the contributor experience even better, and your input is key to achieving excellence. Thank you for helping us improve!

**Pull request merged**